nest/src/token/pUSD.sol

```diff
@@ -6,137 +6,493 @@ import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/I
 import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 import { ERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
-import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ERC4626Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC4626Upgradeable.sol";
+import { ERC20 } from "@solmate/tokens/ERC20.sol";
```

Comment:
Could we change `IAtomicQueue` to accept `IERC20` instead of `ERC20` in the interface? I think we should avoid mixing solmate's `ERC20` import with OpenZeppelin's `ERC20Upgradeable`